src/test/java/org/codehaus/plexus/util/xml/pull/MXParserTest.java

```diff
@@ -391,6 +391,23 @@ public void testSubsequentProcessingInstructionMoreThan8k()
         assertEquals( XmlPullParser.END_TAG, parser.nextToken() );
     }
 
+    @Test
+    public void testFillBuf_NoOverflow()
```

Comment:
Could you write the test in terms of the public API? Accessing directly the fields and calling directy fillBuf() is not the normal use the parser. I'm thinking generating a big enough XML file, parsing it, showing that fails without the patch and pass the test with the patch in place.

Reply:
> Could you write the test in terms of the public API? Accessing directly the fields and calling directy fillBuf() is not the normal use the parser. I'm thinking generating a big enough XML file, parsing it, showing that fails without the patch and pass the test with the patch in place.

Sure https://github.com/codehaus-plexus/plexus-utils/pull/72/commits/a3f9276f4c2ce087d7d34fefadc9142c6eb32601 (without the patch that test fails with `java.io.IOException: error reading input, returned 0`